internal/controller/changetransferpolicy_controller.go

```diff
@@ -360,20 +366,21 @@ func (r *ChangeTransferPolicyReconciler) creatOrUpdatePullRequest(ctx context.Co
 			}
 			err = r.Create(ctx, &pr)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to create PR from branch %q to %q: %q", ctp.Spec.ProposedBranch, ctp.Spec.ActiveBranch, err)
 			}
 			r.Recorder.Event(ctp, "Normal", "PullRequestCreated", fmt.Sprintf("Pull Request %s created", pr.Name))
 			logger.V(4).Info("Created pull request")
 		} else {
-			return err
+			return fmt.Errorf("failed to get PR %q: %w", prName, err)
 		}
 	} else {
-		// Pull request already exists, update it.
-		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Pull Request already exists, update it.
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
 			prUpdated := promoterv1alpha1.PullRequest{}
-			err := r.Get(ctx, client.ObjectKey{Namespace: pr.Namespace, Name: pr.Name}, &prUpdated)
+			// TODO: consider skipping this Get on the first attempt, the object we already got might be up to date.
```

Comment:
Not sure the extra code complexity to do this is worth the cost of hitting the cache again

Reply:
Yeah and are we even hitting the API or just the informer cache?